src/cr/cube/matrix/assembler.py

```diff
@@ -624,6 +629,22 @@ def zscores(self):
             )
         )
 
+    def _assemble_marginal(self, marginal):
+        """Optional 1D ndarray created from a marginal.
+
+        The assembled marginal is the shape of either a row or column (determined by
+        `marginal.direction`), and with the ordering that's applied to those
+        dimensions.
+
+        It is None when the marginal is not defined (`marginal._is_defined`).
+        """
+        if not marginal.is_defined:
+            return None
+
+        order = self._column_order if marginal.direction == "row" else self._row_order
+
+        return np.hstack(marginal.blocks)[order]
```

Comment:
Okay, one thing that occurred to me after reflection. If we do it this `._assemble_marginal()` way, we will still need `._assemble_vector()` or its equivalent because the row and column label vectors are not a marginal and won't need anything outside of `Assembler` to determine their values (they come straight from `Dimension`). So I suppose I'm back to favoring `._assemble_column_vector` and `._assemble_row_vector`, although there could be other options. Why don't you have a noodle on it and we can discuss on the next pass or whatever.

Reply:
I added `_assemble_marginal` and haven't refactored the ones that currently use `_assemble_vector`. But right now, the labels actually already have their own function (so we currently have 3).

- `_assemble_vector` - Used by `rows/columns_margin`, & `rows/columns_margin_proportion`
- `_dimension_labels` - Used by `row/column_labels` (and does the assembling)
- `_assemble_marginal` - My new method

In later refactoring, I think there should be two `_assemble_marginal` and `_assemble_vector`, and internally `_assemble_marginal` should call `_assemble_vector`, it just uses the marginal's information to pass the right arguments.